spec/requests/api/v1/targets/show_spec.rb

```diff
@@ -38,4 +42,19 @@
       expect(response).to have_http_status(:not_found)
     end
   end
+
+  context 'when the user is not authenticated' do
+    let(:headers) { nil }
+    let(:param) { 0 }
+
+    it 'returns the error messages as a json' do
+      subject
+      expect(response.body).to include_json(errors: [I18n.t('devise.failure.unauthenticated')])
+    end
+
+    it 'returns a unauthorized status' do
+      subject
+      expect(response).to have_http_status(:unauthorized)
+    end
+  end
```

Comment:
I think this context could be inside the `'when the request is not valid'` context. WDYT?